cirq-google/cirq_google/study/parameter.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional, Sequence
+import dataclasses
+from typing_extensions import Protocol
+import cirq
+
+
+class SupportsParameter(Protocol):
+    """Protocol for using device parameter keys.
+
+    Args:
+       path: path of the key to modify, with each sub-folder as a string
+           entry in a list.
+       idx: If this key is an array, which index to modify.
+       value: value of the parameter to be set, if any.
+    """
+
+    path: Sequence[str]
+    idx: Optional[int] = None
+    value: Optional[Any] = None
+
+
+@dataclasses.dataclass
+class Parameter(SupportsParameter):
```

Comment:
To prevent future parameter types from colliding, could for this name to be more specific? Maybe `DeviceParameter`?

Reply:
Changed.  Originally chose Parameter to keep compatibility with internal naming, but we should probably convert the internal name too.